java/core/src/main/java/org/bondlib/protocol/SchemaHelper.java

```diff
@@ -0,0 +1,78 @@
+package org.bondlib.protocol;
+
+import org.bondlib.Metadata;
+import org.bondlib.Modifier;
+
+/**
+ * Contains helper methods for working with schema.
+ */
+public final class SchemaHelper {
+
+    private SchemaHelper() {
+    }
+
+    public static boolean isOptionalField(Metadata fieldMetadata) {
+        return fieldMetadata.modifier.value == Modifier.Values.Optional;
+    }
+
+    public static boolean isRequiredField(Metadata fieldMetadata) {
+        return fieldMetadata.modifier.value == Modifier.Values.Required;
+    }
+
+    public static boolean isRequiredOptionalField(Metadata fieldMetadata) {
+        return fieldMetadata.modifier.value == Modifier.Values.RequiredOptional;
+    }
+
+    public static byte getDefaultUInt8FieldValue(Metadata fieldMetadata) {
```

Comment:
Abstraction of the schema so that we don't have to deal with Variant type. The logic is not intuitive and heavily influenced by C++ (e.g. piggy back on int to get bool default, while string/wstring have separate values). 

---
In reply to: [138743899](https://github.com/Microsoft/bond/pull/611#discussion_r138743899) [](ancestors = 138743899)

Reply:
[C# has this logic inlined](https://github.com/Microsoft/bond/blob/master/cs/src/core/protocols/SimpleBinary.cs#L137), and it's exactly as foreign to C# as it is to Java. I don't see a useful abstraction here - you still have to know which of these to call, it's still going to blow up if you call the wrong one.

Reply:
Again, I am following best practices and not C#'s implementation. Knowing which one to call is straightforward while knowing how to get the default value for a field of type bool or int is not. I'd like to keep it as is.